refactor!: remove parameter `df` from section constructors

mbelak-dtml opened the pull request:
BREAKING CHANGE: Parameter `df` is removed from constructors of `GroupAnalysis` and `UnivariateAnalysis`.

lukany reviewed (changes requested):
I'd use `refactor!` type and add `BREAKING CHANGE` footer.

mbelak-dtml commented:
> I'd use `refactor!` type and add `BREAKING CHANGE` footer.

Changed to `refactor!` and added the footer.
Now that I think about it, is this still strictly a refactor if the behavior is slightly changed (`GroupAnalysis` constructor no longer raises when the parameter `groupby` contains an unknown column -- there is no way to check it now)? :thinking: 
I would keep it as a refactor anyway though, since it's breaking anyway and it's an edge case.
